python/snewpy/flux.py

```diff
@@ -0,0 +1,141 @@
+        if(isinstance(axis, str)):
+            axis = Axes[axis]
+        array = np.sum(self.array, axis = axis, keepdims=True)
+        axes = list(self._axes)
+        axes[axis] = axes[axis].take([0,-1])
+        return ContainerClass(array.unit)(array,*axes)
+    
+    def integrate(self, axis:Union[Axes,str], limits:np.ndarray=None)->'Container':
+        """integrate along given axis, producing a reduced array"""
+        if(isinstance(axis, str)):
+            axis = Axes[axis]
+        #set the limits
+        ax = self._axes[axis]
+        xmin, xmax = ax.min(), ax.max()
+        if limits is None:
+            limits = u.Quantity([xmin, xmax])
+        limits = limits.to(ax.unit)
+        limits = limits.clip(xmin, xmax)
+        #compute the integral
+        x  = self._axes[axis]
+        yc = cumulative_trapezoid(self.array, x=x, axis=axis, initial=0)
+        _integral = interp1d(x=x, y=yc, fill_value=0, axis=axis, bounds_error=False)
+        array = np.diff(_integral(limits),axis=axis) << (self.array.unit*ax.unit)
+        axes = list(self._axes)
+        axes[axis] = limits
+        #choose the proper class
+        return ContainerClass(array.unit)(array, *axes)
+
+    def __rmul__(self, factor):
+        "multiply array by givem factor or matrix"
+        return self.__mul__(self, factor)
+
+    def __mul__(self, factor) -> 'Container':
+        "multiply array by givem factor or matrix"
+        #if not (np.isscalar(factor)):
+        #    raise ValueError("Factor should be a scalar value")
+        array = self.array*factor
+        axes = list(self._axes)
+        return ContainerClass(array.unit)(array, *axes)
+
+#a dictionary holding classes for each unit
+_unit_classes = {}
+
+def ContainerClass(Unit, name=None):
+    """Choose appropriate container class for the given unit"""
+    if Unit in _unit_classes:
+        return _unit_classes[Unit]
+        
+    class _cls(Container):
+        unit = Unit
+        def __init__(self, data: u.Quantity,
+                           flavor: np.array,
+                           time: u.Quantity[u.s], 
+                           energy: u.Quantity[u.MeV]
+                    ):
+            
+            super().__init__(data.to(self.unit),flavor,time,energy)
+            
+    if(name):
+        _cls.__name__=name
+    #register unit classes
+    _unit_classes[Unit] = _cls
+    #return the registered class
+    return _unit_classes[Unit]
+
+#some standard container classes that can be used for 
+Flux = ContainerClass(u.one/u.MeV/u.s/u.cm**2, "d2FdEdT")
+Fluence = ContainerClass(Flux.unit*u.s, "dFdT")
+Spectrum= ContainerClass(Flux.unit*u.MeV, "dFdE")
+IntegralFlux = ContainerClass(u.one/u.MeV/u.s/u.cm**2, "F")
+
+DifferrentialEventRate = ContainerClass(u.one/u.MeV/u.s, "d2NdEdT")
+EventRate = ContainerClass(u.one/u.s, "dNdT")
+EventSpectrum = ContainerClass(u.one/u.MeV, "dNdE")
+EventNumbers = ContainerClass(u.one, "N")
```

Comment:
```suggestion
#some standard container classes that can be used for 
Flux = ContainerClass(u.one/u.MeV/u.s/u.cm**2, "d2FdEdT")
Fluence = ContainerClass(Flux.unit*u.s, "dFdE")
Spectrum= ContainerClass(Flux.unit*u.MeV, "dFdT")

DifferentialEventRate = ContainerClass(u.one/u.MeV/u.s, "d2NdEdT")
EventRate = ContainerClass(u.one/u.s, "dNdT")
EventSpectrum = ContainerClass(u.one/u.MeV, "dNdE")
EventNumber = ContainerClass(u.one, "N")
```
* Mixup in Fluence/Spectrum names
* `IntegralFlux` seemed to be redundant to the `Flux` defined a few lines above.
* Typo in “differential”
* Make `EventNumber` singular, for consistency with other container classes